src/lib.rs

```diff
@@ -189,6 +189,7 @@ extern "C" {
 /// The reset handler
 ///
 /// This is the entry point of all programs
+#[link_section = ".text.reset_handler"]
```

Comment:
Q: should we rename this linker section to `.reset_handler`? The name is an implementation detail of the cortex-m-rt as no other crate is supposed to use this section. Now I wonder if `.text.reset_handler` makes more likely that the linker section may get misused. For example some C code could place code in this section if a function is named `reset_handler` and `-ffunction-sections` is used, or something along those lines.